ddl/ddl_api.go

```diff
@@ -1269,6 +1269,12 @@ func (d *ddl) AlterTable(ctx sessionctx.Context, ident ast.Ident, specs []*ast.A
 		return errRunMultiSchemaChanges
 	}
 
+	is := d.infoHandle.Get()
+	t, err := is.TableByName(ident.Schema, ident.Name)
+	if err == nil && t.Meta().IsView() {
```

Comment:
It's weird..

Reply:
Em... actually, I prefer this rather than `TableIsView` function now.
maybe below more understandable @winkyao 
```go
if t != nil && t.Meta().IsView() {
	...
}
```